### PR TITLE
 More precise simulation log icons.

### LINKED ISSUE #167: More precise simulation log icons.
### Describe the solution you'd like

The simulation log currently always shows a default icon for each executed flow node, even if the flow node has a specific type with a more precise icon. See the following example:

image

Here, one could show a receive task icon and event-specific icons for the other events. The icons exist in the BPMN-font and could be used by changing [Log.js](https://github.com/bpmn-io/bpmn-js-token-simulation/blob/main/lib/features/log/Log.js).

If you are interested in this change, I can draft a pull request since I implemented this in my tool based on the token simulation.

Changing to more precise icons can help distinguish two executed flow nodes without names and make the simulation log consistent with the icons in the model.

### Additional context

Currently, it looks like this:
image
More precise icons would look like this:
image

### PR DESCRIPTION
**Pull request by timKraeuter**
More precise simulation log icons (Closes #167).

I didn't add any tests since there were none before, but I could do so if needed. I used the following models to test the improved logging manually:
[intermediate_events.txt](https://github.com/bpmn-io/bpmn-js-token-simulation/files/14635569/intermediate_events.txt)
[end_events.txt](https://github.com/bpmn-io/bpmn-js-token-simulation/files/14635570/end_events.txt)
[start_events.txt](https://github.com/bpmn-io/bpmn-js-token-simulation/files/14635571/start_events.txt)
[tasks.txt](https://github.com/bpmn-io/bpmn-js-token-simulation/files/14635572/tasks.txt)

I added two TODOS for other broken/missing functionality:
1. There are **no traces for terminate end events**. Probably since the process is aborted.
2. There is code that tries to **log which outgoing sequence flow is chosen** for an exclusive gateway, but this does not work when I tried to test it.

### CODE REVIEW

File: lib/features/log/Log.js [hunk 1]
```diff
@@ -151,96 +198,142 @@ export default function Log(
...
+    if (is(element, 'bpmn:IntermediateThrowEvent')) {
+      this.log({
+        text: elementName || 'Intermediate Event',
+        icon: getIconForIntermediateEvent(element, 'throw'),
+        scope
+      });
+      return;
+    }
+    if (is(element, 'bpmn:BoundaryEvent')) {
+      this.log({
```
Comment (nikku) [inline lib/features/log/Log.js:324]:
```suggestion
      return this.log({

```

If we apply an eager return instead of `if/else` nesting, let's do it like this :arrow_up:

Comment (timKraeuter) [reply lib/features/log/Log.js:324]: We can do that, but why do you prefer that?
We should then probably do it consistently for all cases.

Comment (nikku) [reply lib/features/log/Log.js:324]: Yes of course. If we drop nesting, let's establish

```
return iDidWork()
```

as a pattern.

Comment (timKraeuter) [reply lib/features/log/Log.js]: Just changed it.

**Comment by nikku** [comment]
Logged what you reported as a bug (termination) [here](https://github.com/bpmn-io/bpmn-js-token-simulation/issues/169).

**Comment by nikku** [comment]
A test looks like [this](https://github.com/bpmn-io/bpmn-js-token-simulation/blob/main/test/spec/SimulationSpec.js#L59), asserting the path. You trigger tests through the [`triggerElement` utility](https://github.com/bpmn-io/bpmn-js-token-simulation/blob/main/test/spec/SimulationSpec.js#L63). 

I suggest that we replace the `Log#log` component to capture actual log statements, and assert what was logged:

```

class CustomLog extends Log {
  constructor(...) {
    super(...);
    
    this._logged = [];
    this.log = (args) => {
      this._logged.push(args);
    };
    
    ...
  }
}

beforeEach(bootstrapBpmnModeler({
  additionalModules: [ {
    log: [ 'type', CustomLog ]
  } ]
});
```

**Comment by timKraeuter** [comment]
Just added the tests. Let me know what you think.

**Comment by timKraeuter** [comment]
> > > Logged what you reported as a bug (termination) #169.
> > 
> > 
> > What do you think about my 2nd point? Is it worth it to change the trace event to include the chosen sf for an exclusive gateway and fix that part?
> 
> We _could_ consider to log sequence flows (or message flows) taken. Ultimately this depends on whether it is information overkill or not. At some point a backtrace (understand where a token came from) may be a better visualization.

Alright, I will remove the dead code for now. I also saw we are not logging parallel gateways. Should we do that to stay consistent with exclusive gateways or is it just information overload?

**Comment by nikku** [comment]
I think we should consistently log all flow nodes and _may_ consider to log sequence and message flows in the future.

**Comment by timKraeuter** [comment]
I will change that and add one more test for gateways. Inclusive gateways also need to be included.

**Comment by timKraeuter** [comment]
The changes should be ready for review now.

**Comment by nikku** [comment]
Thanks for the follow-up. I pushed a bunch of style changes in top. Great contribution + great test coverage added. :rocket:
